pkg/crane/crane_test.go

```diff
@@ -282,6 +293,9 @@ func TestBadInputs(t *testing.T) {
 		crane.Copy(invalid, invalid),
 		crane.Copy(valid, invalid),
 		crane.Copy(valid, valid), // 404
+		crane.Tag(invalid, invalid),
+		crane.Tag(valid, invalid),
+		crane.Tag(valid, valid), // 404
```

Comment:
It's not immediately apparent to me why this should return a 404. Is it because `valid` (while valid) doesn't exist? Maybe a better name would help?

Also, I noticed that if any of these don't return an error, the test failure message is going to be pretty unhelpful. Can you make this a `[]struct{desc string, err error}` instead, and at least log which case resulted in the nil error?

Reply:
Done